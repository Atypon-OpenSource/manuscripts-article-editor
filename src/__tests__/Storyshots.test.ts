jest.mock('../lib/token')
jest.mock('../lib/deviceId')
jest.mock('react-modal')

import initStoryshots, { renderOnly } from '@storybook/addon-storyshots'

initStoryshots({
  configPath: 'stories/config',
  storyKindRegex: /^(Application Menu|Authentication|Avatar|Button|Collaborators.*|Cross References|Dropdown|Footer|Forms.*|Hero|Invitation|Menu|MenuBar|Page|PageHeading|Pages|Panel|Projects Dropdown List|SelectField|TextField.*)$/,
  test: renderOnly,
})
