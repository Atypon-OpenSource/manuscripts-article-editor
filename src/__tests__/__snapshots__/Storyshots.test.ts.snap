// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Button with text 1`] = `
<div
  className="sc-bdVaJa jrlBps"
>
  <div>
    <button
      className="sc-bxivhb igRzyj"
    >
      Hello Button
    </button>
  </div>
</div>
`;

exports[`Storyshots IconButton with text 1`] = `
<div
  className="sc-bdVaJa jrlBps"
>
  <div>
    <button
      className="sc-EHOje iHIijj"
      onClick={[Function]}
    >
      <svg
        height={undefined}
        width={undefined}
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <defs>
          <filter
            filterUnits="objectBoundingBox"
            height="200%"
            id="a"
            width="200%"
            x="-50%"
            y="-50%"
          >
            <feOffset
              dy={1}
              in="SourceAlpha"
              result="shadowOffsetOuter1"
            />
            <feGaussianBlur
              in="shadowOffsetOuter1"
              result="shadowBlurOuter1"
              stdDeviation={0.5}
            />
            <feColorMatrix
              in="shadowBlurOuter1"
              result="shadowMatrixOuter1"
              values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.168 0"
            />
            <feOffset
              in="SourceAlpha"
              result="shadowOffsetOuter2"
            />
            <feGaussianBlur
              in="shadowOffsetOuter2"
              result="shadowBlurOuter2"
              stdDeviation={0.5}
            />
            <feColorMatrix
              in="shadowBlurOuter2"
              result="shadowMatrixOuter2"
              values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.084 0"
            />
            <feMerge>
              <feMergeNode
                in="shadowMatrixOuter1"
              />
              <feMergeNode
                in="shadowMatrixOuter2"
              />
              <feMergeNode
                in="SourceGraphic"
              />
            </feMerge>
          </filter>
          <rect
            height={40}
            id="b"
            rx={2}
            width={40}
          />
        </defs>
        <g
          fill="none"
          fillRule="evenodd"
        >
          <g
            filter="url(#a)"
            transform="translate(3 3)"
          >
            <use
              fill="#FFF"
              xlinkHref="#b"
            />
            <use
              xlinkHref="#b"
            />
            <use
              xlinkHref="#b"
            />
            <use
              xlinkHref="#b"
            />
          </g>
          <path
            d="M31.64 23.205c0-.639-.057-1.252-.164-1.841H23v3.481h4.844a4.14 4.14 0 0 1-1.796 2.716v2.259h2.908c1.702-1.567 2.684-3.875 2.684-6.615z"
            fill="#4285F4"
          />
          <path
            d="M23 32c2.43 0 4.467-.806 5.956-2.18l-2.908-2.259c-.806.54-1.837.86-3.048.86-2.344 0-4.328-1.584-5.036-3.711h-3.007v2.332A8.997 8.997 0 0 0 23 32z"
            fill="#34A853"
          />
          <path
            d="M17.964 24.71a5.41 5.41 0 0 1-.282-1.71c0-.593.102-1.17.282-1.71v-2.332h-3.007A8.996 8.996 0 0 0 14 23c0 1.452.348 2.827.957 4.042l3.007-2.332z"
            fill="#FBBC05"
          />
          <path
            d="M23 17.58c1.321 0 2.508.454 3.44 1.345l2.582-2.58C27.463 14.891 25.426 14 23 14a8.997 8.997 0 0 0-8.043 4.958l3.007 2.332c.708-2.127 2.692-3.71 5.036-3.71z"
            fill="#EA4335"
          />
          <path
            d="M14 14h18v18H14V14z"
          />
        </g>
      </svg>
    </button>
  </div>
</div>
`;

exports[`Storyshots LinkButton with text 1`] = `
<div
  className="sc-bdVaJa jrlBps"
>
  <div>
    <a
      className="sc-gzVnrw fjbzlL"
      href="/example"
      onClick={[Function]}
    >
      primary
    </a>
  </div>
</div>
`;

exports[`Storyshots PlainButton with text 1`] = `
<div
  className="sc-bdVaJa jrlBps"
>
  <div>
    <button
      className="sc-bZQynM kpcFGy"
      onClick={[Function]}
    >
      primary
    </button>
  </div>
</div>
`;

exports[`Storyshots PrimaryButton with text 1`] = `
<div
  className="sc-bdVaJa jrlBps"
>
  <div>
    <button
      className="sc-ifAKCX EnKZR"
      onClick={[Function]}
    >
      primary
    </button>
  </div>
</div>
`;
